Documents Management Policies (#407)
crossplane/docs: content/master/concepts/managed-resources.md

this example, creation call of Azure MySQL Server will not be made until
referenced `ResourceGroup` has its `status.condition` named `Ready` to be true.

## Management Policies

Crossplane offers a set of management policies that allow you to define the
level of control it has over external resources. You can configure these
policies using the `spec.managementPolicy` field in the managed resource
definition. The available policies include:

- `FullControl (Default)`: With this policy, Crossplane fully manages and
controls the external resource.
- `ObserveOnly`: With the ObserveOnly policy, Crossplane only observes the

> external resource without making any changes or deletions.

Reviewer: Just a comment to confirm.

So we are agreed on dropping OrphanOnDelete and keeping the DeletionPolicy. Im OK with that no matter what is the end result of the ignore changes proposal.

creation of the managed resource will be rejected. So, in those cases, you will
need to enter the name of the resource as well as the required fields.

### Alternative Import Procedure: Start with ObserveOnly

The above approach has the following caveats:

1. You need to provide all the required fields in the spec of the resource with
correct values even though they are not used for importing the resource. A wrong

> value for a required field will result a configuration update which is not

Reviewer: ```suggestion
value for a required field results in a configuration update which is not
```

desired.

> 2. Any typo in the external name annotation or some mistake in the identifying
> arguments (e.g. `region`) will result creation of a new resource instead of
> importing the existing one.

Reviewer: ```suggestion
2. Any typos in the external name annotation or mistakes in the identifying arguments, such as the region, results in the creation of a new resource instead of importing the existing one.
```

> Alternatively, you can import with first observing the resource with
> management policy `ObserveOnly` by following the procedure below:

Reviewer: ```suggestion
Alternatively, you can import the resource by first observing it with a management policy of `ObserveOnly`. To do this, follow the procedure below:
```

1. Create a new resource with `ObserveOnly` policy.
  1. With external name annotation set to the external name of the resource to be
     imported.
  2. Only provide the identifying arguments (e.g. `region`) in the spec of the
     resource.
2. Expect the existing resource to be observed successfully indicating that the
   existing resource is found.
3. Change the policy to `FullControl` and provide the required fields by copying
   them from `status.atProvider` to give full control of the resource to
   Crossplane.

> Please note that management policies is an experimental feature needs to be
> enabled first. See the [management policies] section for more details.

Reviewer: ```suggestion
Please note that management policies are an experimental feature need to be
enabled first. See the [management policies] section for more details.
```

> enabled first. See the [management policies] section for more details.

Author: @plumbis updated as requested.

Previously, I hesitated to add examples as they could bloat the page. Added now; please let me know if this is the case.

creation of the managed resource will be rejected. So, in those cases, you will
need to enter the name of the resource as well as the required fields.

### Alternative Import Procedure: Start with ObserveOnly

Directly importing existing managed resources approach has the following caveats:

1. You must provide all the required fields in the spec of the resource with
correct values even though they're not used for importing the resource. A wrong
value for a required field result in a configuration update which isn't
desired.
2. Any typos in the external name annotation or mistakes in the identifying
arguments, such as the `region`, results in the creation of a new resource
instead of importing the existing one.

Instead of manually creating resources you can import the resource with an
`ObserveOnly` management policy.

Crossplane imports `ObserveOnly` resources but never changes or deletes the
resource.

{{< hint "important" >}}
Management policies including `ObserveOnly` are experimental. They must be
explicitly enabled.
See the management policies section for more details.
{{< /hint >}}

To configure an `ObserveOnly` resource:

> 1. Create a new resource with an `ObserveOnly` management policy.
>    1. With external name annotation set to the external name of the resource to be
>       imported.
>    2. Only provide the identifying arguments (for example, `region`) in the spec
>       of the resource.
> 
> ```yaml

Reviewer 2: ```suggestion
1. Create a new resource with an {{<hover label="oo" line="8">}}ObserveOnly{{</hover>}} 
management policy.
   1. With the 
   {{<hover label="oo" line="5">}}crossplane.io/external-name{{</hover>}} 
   annotation set to the external name of the resource to import.
   1. Only provide the identifying arguments (for example, 
   {{<hover label="oo" line="10">}}region{{</hover>}}) in the spec
      of the resource.

```yaml {label="oo"}
```

Adds the [`hover` shortcode](https://docs.crossplane.io/contribute/code-style-guide/#dynamic-line-highlighting) to allow readers to hover over the command and see where it is in the yaml example.

apiVersion: sql.gcp.upbound.io/v1beta1
kind: DatabaseInstance
metadata:
  annotations:
    crossplane.io/external-name: existing-database-instance
  name: existing-database-instance
spec:
  managementPolicy: ObserveOnly
  forProvider:
    region: "us-central1"
```

> Crossplane discovers the managed resource and populates the `status.atProvider`
> with the observed state.
> 
> ```yaml

Reviewer 2: ```suggestion
Crossplane discovers the managed resource and populates the 
{{<hover label="ooPopulated" line="12">}}status.atProvider{{</hover>}}
with the observed state.

```yaml {label="ooPopulated"}
```
Adds hover shortcode.

apiVersion: sql.gcp.upbound.io/v1beta1
kind: DatabaseInstance
metadata:
  annotations:
    crossplane.io/external-name: existing-database-instance
  name: existing-database-instance
spec:
  managementPolicy: ObserveOnly
  forProvider:
    region: us-central1
status:
  atProvider:
    connectionName: crossplane-playground:us-central1:existing-database-instance
    databaseVersion: POSTGRES_14
    deletionProtection: true
    firstIpAddress: 35.184.74.79
    id: existing-database-instance
    publicIpAddress: 35.184.74.79
    region: us-central1
    <truncated-for-brevity>
    settings:
    - activationPolicy: ALWAYS
      availabilityType: REGIONAL
      diskSize: 100
      <truncated-for-brevity>
      pricingPlan: PER_USE
      tier: db-custom-4-26624
      version: 4
  conditions:
  - lastTransitionTime: "2023-02-22T07:16:51Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "2023-02-22T07:16:51Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
```

To allow Crossplane to control and change the `ObserveOnly` resource, edit the

> policy. Change the `ObserveOnly` field to `FullControl`. If there are required
> parameters, copy their values from `status.atProvider` and provide them at the
> `spec.forProvider`.
> 
> ```yaml

Reviewer 2: ```suggestion
policy.  

Change the {{<hover label="ooPopulated" line="8">}}ObserveOnly{{</hover>}} field 
to {{<hover label="fc" line="8">}}FullControl{{</hover>}}. 

Copy any required parameter values from 
{{<hover label="fc" line="16">}}status.atProvider{{</hover>}} and provide them 
in {{<hover label="fc" line="9">}}spec.forProvider{{</hover>}}.

```yaml {label="fc"}
```

Adds the hover shortcode and adds line breaks for readability.
